src/BenchmarksApps/TodosApi/ValidationFilter.cs

```diff
@@ -0,0 +1,166 @@
+    }
+
+    public ValueTask<object?> InvokeAsync(EndpointFilterInvocationContext context, EndpointFilterDelegate next)
+    {
+        var validatableParametersMap = context.HttpContext.GetEndpoint()?.Metadata.OfType<ValidationMetadata>()
+            .FirstOrDefault()?.ValidatableParametersMap;
+
+        if (validatableParametersMap is null)
+        {
+            return next(context);
+        }
+
+        Dictionary<string, List<string>>? workingErrors = null;
+
+        foreach (var index in validatableParametersMap)
+        {
+            var validatable = context.GetArgument<IValidatableObject?>(index);
+            if (validatable is not null)
+            {
+                var requiresServices = validatable is IValidatable v && v.RequiresServiceProvider;
+                var validationContext = GetValidationContext(validatable, requiresServices ? context.HttpContext.RequestServices : null);
+                if (_logger.IsEnabled(LogLevel.Information))
+                {
+                    _logger.LogInformation("Validating argument at index {index} of type {type}", index, validatable.GetType().Name);
+                }
+                var results = validatable.Validate(validationContext);
+
+                if (results.Any())
+                {
+                    if (_logger.IsEnabled(LogLevel.Information))
+                    {
+                        _logger.LogInformation("Argument at index {index} of type {type} has validation errors", index, validatable.GetType().Name);
+                    }
+                    workingErrors ??= new();
+
+                    foreach (var result in results)
+                    {
+                        foreach (var member in result.MemberNames)
+                        {
+                            List<string>? messages;
+                            if (!workingErrors.ContainsKey(member))
+                            {
+                                messages = new();
+                                workingErrors.Add(member, messages);
+                            }
+                            else
+                            {
+                                messages = workingErrors[member];
+                            }
+                            messages.Add(result.ErrorMessage ?? "The value provided was invalid.");
+                        }
+                    }
+
+                    var errors = MapToFinalErrorsResult(workingErrors);
+                    var problem = TypedResults.ValidationProblem(errors);
+                    return ValueTask.FromResult((object?)problem);
+                }
+            }
+        }
+        return next(context);
+    }
+
+    [UnconditionalSuppressMessage("Trimming", "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code",
```

Comment:
```suggestion
    [UnconditionalSuppressMessage("Trimming", "IL2026:RequiresUnreferencedCode",
```

This is shorter, and conveys the same amount of information.